OSDOCS-10889 OCP 4.14.30 Release Notes (#77589)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-14-30"]
=== RHSA-2024:3881 - {product-title} 4.14.30 bug fix and security updates

Issued: 2024-06-19

{product-title} release 4.14.30, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:3881[RHSA-2024:3881] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:3918[RHSA-2024:3918] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.30 --pullspecs
----

[id="ocp-4-14-30-bug-fixes"]
==== Bug fixes

> * Previously,  (link:https://issues.redhat.com/browse/OCPBUGS-35085[*OCPBUGS-35085*])

Reviewer: * Previously, the OpenShift Agent Installer reported installed SATA SDDs as removable and refused to use any of them as installation targets. With this release removable disks are eligible for installation and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-35085[*OCPBUGS-35085*])

> * https://issues.redhat.com/browse/OCPBUGS-35034

Reviewer: Enhancement!

[id="ocp-4-15-18-password-colon-character"]
===== Allowing pull secret passwords to contain a colon character

* This release adds the ability to include a colon character in your password for the Openshift Assisted Installer.

* Previously, an AWS policy issue prevented the Cluster API Provider for AWS from retrieving the necessary domain information. As a consequence, installing an AWS hosted cluster with a custom domain failed. With this update, the policy issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34856[*OCPBUGS-34856*])

> * https://issues.redhat.com/browse/OCPBUGS-34814

Reviewer: * Previously, when a cluster or BareMetal Host (BMH) was deleted, a `PreprovImage` image was created during the _powering off before delete_ stage. As a result, the cluster resource got stuck. With this release, an exception was created for the _powering off before delete_ stage and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34814[OCPBUGS-34814]).

> * https://issues.redhat.com/browse/OCPBUGS-34792 RHE

Reviewer: Still looks private here

> * https://issues.redhat.com/browse/OCPBUGS-34668

Reviewer: Enhancement:
Previously, when `virtualHostedStyle` with the `regionEndpoint` was set in the image registry operator config, the image registry ignored the virtual hosted style config and failed to start. This release adds a new upstream distribution configuration (force-path style) in favor of the downstream-only version (virtual-hosted style). (link:https://issues.redhat.com/browse/OCPBUGS-34668[OCPBUGS-34668])

> * https://issues.redhat.com/browse/OCPBUGS-34657

Reviewer: * Previously, the 'metal3-ironic' and 'metal3-ironic-inspector' pods failed when upgrading to {product-title} 4.15.11 from 4.15.8 due to an install failure related to FIPS mode enablement. With this release, the issue has been resolved. (link:[https://issues.redhat.com/browse/OCPBUGS-34657[OCPBUGS-34657]).

* Previously, certain situations caused transfer of an Egress IP address from one node to a different node to fail, and this failure impacted the OVN-Kubernetes network. The network failed to send gratuitous Address Resolution Protocol (ARP) requests to peers to inform them of the new node’s medium access control (MAC) address. As a result, peers would temporarily send reply traffic to an old node and this traffic led to failover issues. With this release, the OVN-Kubernetes network correctly sends a gratuitous ARP to peers to inform them of the new Egress IP node MAC address, so that each peer can send reply traffic to the new node without causing failover time issues. (link:https://issues.redhat.com/browse/OCPBUGS-34570[*OCPBUGS-34570*])

> * Previously, the `wait-for-ceo` command used during the bootstrapping process to verify etcd rollout would not report errors for some failure modes. With this release, those error messages should now be visible on the `bootkube` script if the command exits in an error case. (link:https://issues.redhat.com/browse/OCPBUGS-34495[*OCPBUGS-34495*])

Reviewer: Replace "should" with "are"

> * https://issues.redhat.com/browse/OCPBUGS-5943

Reviewer: * Previously, the DNS Operator did not verify that a cluster had ready nodes with available CPU in at least two availability zones, and the DNS daemon set did not use surge for rolling updates. As a result, clusters in which all nodes were in the same availability zone would repeatedly emit `TopologyAwareHintsDisabled` events for the cluster DNS service. With this release, the `TopologyAwareHintsDisabled` events are no longer emitted on clusters that do not have nodes in multiple availability zones and the issue has been resolved. (link:[https://issues.redhat.com/browse/OCPBUGS-5943[OCPBUGS-5943])

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-14-30"]
=== RHSA-2024:3881 - {product-title} 4.14.30 bug fix and security updates

Issued: 2024-06-19

{product-title} release 4.14.30, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:3881[RHSA-2024:3881] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:3918[RHSA-2024:3918] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.30 --pullspecs
----

[id="ocp-4-14-30-enhancements"]
==== Enhancements

> The following enhancement is included in this z-stream release:

Reviewer 2: ```suggestion
The following enhancements are included in this z-stream release:
```

[id="ocp-4-15-30-password-colon-character"]
===== Allowing a colon character in pull secret passwords

This release adds the ability to include a colon character in your pull secret password for the {product-title} {ai-full}. (link:https://issues.redhat.com/browse/OCPBUGS-35034[*OCPBUGS-35034*])

[id="ocp-4-15-30-csi-topology-awareness"]
===== Configuring the topology feature for the Cinder CSI Driver

Previously, the topology feature for the Cinder CSI Driver was always active, because you could not disable the topology feature. With this release, the topology feature is based on the compute and storage availability zones and you can disable the topology feature. (link:https://issues.redhat.com/browse/OCPBUGS-34792[*OCPBUGS-34792*])

[id="ocp-4-14-30-bug-fixes"]
==== Bug fixes

* Previously, the {product-title} {ai-full} reported installed SATA SDDs as removable and did not use any of them as installation targets. With this release, removable disks are eligible for installation and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-35085[*OCPBUGS-35085*])

> * Previously, an Amazon Web Services (AWS) policy issue prevented the Cluster API Provider for AWS from retrieving the necessary domain information. As a consequence, installing an AWS hosted cluster with a custom domain failed. With this update, the policy issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34856[*OCPBUGS-34856*])

Reviewer 2: Normally would not hold this for the use of attributes, but the `Cluster API Provider AWS` one replaces incorrect text (there is no "for" in the phrase)
```suggestion
* Previously, an {aws-first} policy issue prevented the {cap-aws-short} from retrieving the necessary domain information. As a consequence, installing an {aws-short} hosted cluster with a custom domain failed. With this update, the policy issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34856[*OCPBUGS-34856*])
```

> * Previously, when a cluster or BareMetal Host (BMH) was deleted, a `PreprovImage` image was created during the powering off before delete stage. As a result, the cluster resource got stuck. With this release, an exception is created for the powering off before delete stage and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34814[*OCPBUGS-34814*]).

Reviewer 3: ```suggestion
* Previously, when you deleted a cluster or BareMetal Host (BMH), a `PreprovImage` image was created during cluster deletion. As a result, the cluster resources were stuck. With this release, an exception is created for the powering off before delete stage and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34814[*OCPBUGS-34814*]).
```

- Is there any particular state, cluster resource were stuck in?
- Can we explain what exception is given for powering off?

> * Previously, when you deleted a cluster or BareMetal Host (BMH), a `PreprovImage` image was created during cluster deletion. As a result, the cluster resources were stuck. With this release, an exception is created for the powering off before delete stage and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34814[*OCPBUGS-34814*]).

Reviewer 2: ```suggestion
* Previously, when you deleted a cluster or BareMetal Host (BMH), a `PreprovImage` image was created during cluster deletion. As a result, the cluster resources were stuck. With this release, an exception is created for the powering off before the delete stage and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34814[*OCPBUGS-34814*]).
```

> * Previously, when `virtualHostedStyle` with the `region` endpoint was set in the image registry operator config, the image registry ignored the virtual hosted style config and failed to start. This release adds a new upstream distribution configuration (force-path style) in favor of the downstream-only version (virtual-hosted style). (link:https://issues.redhat.com/browse/OCPBUGS-34668[*OCPBUGS-34668*])

Reviewer 3: ```suggestion
* Previously, when you enabled `virtualHostedStyle` with `regionEndpoint` in the Image Registry Operator configuration, the image registry ignored `virtualHostedStyle` and failed to start. This release adds a new upstream distribution configuration in favor of the downstream-only version. (link:https://issues.redhat.com/browse/OCPBUGS-34668[*OCPBUGS-34668*])
```

Reviewer 3: - Is `virtualHostedStyle` a parameter we can specify that in the text.
- I am not sure what this means "This release adds a new upstream distribution configuration in favor of the downstream-only version." 
- Please confirm the values in back ticks. For example, if it's `region` endpoint or `regionEndpoint`

* Previously, the `metal3-ironic` and `metal3-ironic-inspector` pods failed when upgrading to {product-title} 4.15.11 from 4.15.8 due to an install failure related to FIPS mode enablement. With this release, the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34657[*OCPBUGS-34657*]).

> * Previously, certain situations caused transfer of an Egress IP address from one node to a different node to fail, and this failure impacted the OVN-Kubernetes network. The network failed to send gratuitous Address Resolution Protocol (ARP) requests to peers to inform them of the new node’s medium access control (MAC) address. As a result, peers would temporarily send reply traffic to an old node and this traffic led to failover issues. With this release, the OVN-Kubernetes network correctly sends a gratuitous ARP to peers to inform them of the new Egress IP node MAC address, so that each peer can send reply traffic to the new node without causing failover time issues. (link:https://issues.redhat.com/browse/OCPBUGS-34570[*OCPBUGS-34570*])

Reviewer 3: I have given my suggestion, but this might need revising.

* Previously, if you used the `wait-for-ceo` command during the bootstrapping process, the command would not report the error messages in case of a failure. With this release,  the command reports the error messages in the `bootkube` script for you to view. (link:https://issues.redhat.com/browse/OCPBUGS-34495[*OCPBUGS-34495*])

> * Previously, the installation program did not support the `ca-west-1` Amazon Web Services (AWS) region. With this release, the `ca-west-1` region is supported and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34024[*OCPBUGS-34024*])

Reviewer 2: ```suggestion
* Previously, the installation program did not support the `ca-west-1` {aws-first} region. With this release, the `ca-west-1` region is supported and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-34024[*OCPBUGS-34024*])
```
